client/state/support/actions.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SUPPORT_USER_ACTIVATED,
+	SUPPORT_USER_DEACTIVATED,
+	SUPPORT_USER_FETCH_TOKEN,
+} from 'state/action-types';
+
+import User from 'lib/user';
+
+import userSettings from 'lib/user-settings';
+
+/**
+ * Requests a support user token, then dispatches the relevant actions upon response
+ * 
+ * @param  {string} supportUser     Support username
+ * @param  {string} supportPassword Support password
+ * @return {thunk}                  The action thunk
+ */
+export function supportUserFetchToken( supportUser, supportPassword ) {
+	return ( dispatch ) => {
+		if ( !supportUser || !supportPassword ) {
+			return;
+		}
+
+		dispatch( {
+			type: SUPPORT_USER_FETCH_TOKEN,
+			supportUser
+		} );
+
+		const user = new User();
+
+		const activateSupportUser = ( userData ) => 
+			dispatch( supportUserActivated( Object.assign( {}, user.data ) ) );
+
+		const tokenErrorCallback = ( error ) => 
+			dispatch( supportUserDeactivated( error.message ) );
+
+		const changeUserCallback = error => error
+			? tokenErrorCallback( error )
+			: activateSupportUser();
+
+		user.changeUser( supportUser, supportPassword, changeUserCallback, tokenErrorCallback );
+	}
+}
+/**
+ * Logs out of support user, restoring the original user
+ * @return {object} The action object
+ */
+export function supportUserRestore() {
+	let user = new User();
+	user.clear();
+	user.restoreUser();
```

Comment:
There's quite a few side-effects occurring in the functions of this file. The action creators should really only be returning simple objects (or in the case of thunks, dispatching objects as asynchronously). It seems that these are needed for interoperability between the old-style user store, though I'd argue that we should start to manage the behavior of the current user exclusively from [`client/state/current-user`](https://github.com/Automattic/wp-calypso/tree/master/client/state/current-user).